File: messageDotNet/grpc/MethodData/HomeData.cs
using System;
using UnityEngine;

/// <summary>
/// call main service save user home data
/// </summary>
[Serializable]
public class MainServiceActionSaveHomeDataInput
{
    public long UserId;
    public GrpcHomeData Data;
}
[Serializable]
public class MainServiceActionSaveHomeDataOutput
{
    public bool Success;
    public string ErrMsg;
  
}

/// <summary>
/// call main service update  user home data last save time
/// </summary>
[Serializable]
public class MainServiceActionUpHomeLastSaveTimeInput
{
    public long UserId;
    public long LastSaveMs; 
}
[Serializable]
public class MainServiceActionUpHomeLastSaveTimeOutput
{
    public bool Success;
    public string ErrMsg; 
}

/// <summary>
/// select user home data for main service
/// </summary>
[Serializable]
public class MainServiceActionGetHomeDataInput
{
    public long UserId; 
}
[Serializable]
public class MainServiceActionGetHomeDataOutput
{
    public bool Success;
    public string ErrMsg;
    public long UserId;
    public GrpcHomeData Data; 

}

/// <summary>
/// call main service save animal list data
/// </summary>
[Serializable]
public class MultiUpdateAnimalBaseDataInput
{
    public long UserId;
    public GrpcAnimalBaseData[] Animals; 
}
[Serializable]
public class MultiUpdateAnimalBaseDataOutput
{
    public bool Success;
    public string ErrMsg; 

}

/// <summary>
///  GetUserAnimalList  world to main service
/// </summary>
[Serializable]
public class GetUserAnimalListInput
{
    public long UserId; 
}
[Serializable]
public class GetUserAnimalListOutput
{
    public bool Success;
    public string ErrMsg;
    public long UserId;
    public GrpcAnimalBaseData[] Animals;
 
}

/// <summary>
///  GetUserAnimalList  world to main service
/// </summary>
[Serializable]
public class CaptureAnimalInput
{
    public long UserId;
    public long FreedAnimalId;
    public GrpcAnimalBaseData Animal;
     
}
[Serializable]
public class CaptureAnimalOutput
{
    public bool Success;
    public string ErrMsg;
}

File: messageGo/grpc/methodData/home_data.go
package methodData

import (
	base_data "game-message-core/grpc/baseData"
)

// call main service save user home data
type MainServiceActionSaveHomeDataInput struct {
	UserId int64                  `json:"userId"`
	Data   base_data.GrpcHomeData `json:"data"`
}

func (p *MainServiceActionSaveHomeDataInput) Clear() {
	p.UserId = 0
	p.Data.Clear()
}

type MainServiceActionSaveHomeDataOutput struct {
	Success bool   `json:"success"`
	ErrMsg  string `json:"errMsg"`
}

func (p *MainServiceActionSaveHomeDataOutput) Clear() {
	p.Success = false
	p.ErrMsg = ""
}

// call main service update  user home data last save time
type MainServiceActionUpHomeLastSaveTimeInput struct {
	UserId     int64 `json:"userId"`
	LastSaveMs int64 `json:"lastSaveMs"`
}

func (p *MainServiceActionUpHomeLastSaveTimeInput) Clear() {
	p.UserId = 0
	p.LastSaveMs = 0
}

type MainServiceActionUpHomeLastSaveTimeOutput struct {
	Success bool   `json:"success"`
	ErrMsg  string `json:"errMsg"`
}

func (p *MainServiceActionUpHomeLastSaveTimeOutput) Clear() {
	p.Success = false
	p.ErrMsg = ""
}

// call main service get user home data
type MainServiceActionGetHomeDataInput struct {
	UserId int64 `json:"userId"`
}

func (p *MainServiceActionGetHomeDataInput) Clear() {
	p.UserId = 0
}

type MainServiceActionGetHomeDataOutput struct {
	Success bool                   `json:"success"`
	ErrMsg  string                 `json:"errMsg"`
	UserId  int64                  `json:"userId"`
	Data    base_data.GrpcHomeData `json:"data"`
}

func (p *MainServiceActionGetHomeDataOutput) Clear() {
	p.Success = false
	p.ErrMsg = ""
	p.UserId = 0
	p.Data.Clear()
}

// call main service save animal list data
type MultiUpdateAnimalBaseDataInput struct {
	UserId  int64                          `json:"userId"`
	Animals []base_data.GrpcAnimalBaseData `json:"animals"`
}

func (p *MultiUpdateAnimalBaseDataInput) Clear() {
	p.UserId = 0
	if p.Animals != nil {
		p.Animals = p.Animals[:0]
	}
}

type MultiUpdateAnimalBaseDataOutput struct {
	Success bool   `json:"success"`
	ErrMsg  string `json:"errMsg"`
}

func (p *MultiUpdateAnimalBaseDataOutput) Clear() {
	p.Success = false
	p.ErrMsg = ""
}

//  GetUserAnimalList  world to main service
type GetUserAnimalListInput struct {
	UserId int64 `json:"userId"`
}

func (p *GetUserAnimalListInput) Clear() {
	p.UserId = 0
}

type GetUserAnimalListOutput struct {
	Success bool                           `json:"success"`
	ErrMsg  string                         `json:"errMsg"`
	UserId  int64                          `json:"userId"`
	Animals []base_data.GrpcAnimalBaseData `json:"animals"`
}

//  capture animal  scene world to main service
type CaptureAnimalInput struct {
	UserId        int64                        `json:"userId"`
	FreedAnimalId int64                        `json:"freedAnimalId"`
	CaptureAnimal base_data.GrpcAnimalBaseData `json:"animal"`
}

func (p *CaptureAnimalInput) Clear() {
	p.UserId = 0
	p.FreedAnimalId = 0
	p.CaptureAnimal.Clear()
}

type CaptureAnimalOutput struct {
	Success bool   `json:"success"`
	ErrMsg  string `json:"errMsg"`
}

func (p *CaptureAnimalOutput) Clear() {
	p.Success = false
	p.ErrMsg = ""
}
